llvm/lib/Target/RISCV/GISel/RISCVCallLowering.cpp

```diff
@@ -246,16 +256,29 @@ struct RISCVIncomingValueHandler : public CallLowering::IncomingValueHandler {
   unsigned assignCustomValue(CallLowering::ArgInfo &Arg,
                              ArrayRef<CCValAssign> VAs,
                              std::function<void()> *Thunk) override {
+    const CCValAssign &VA = VAs[0];
+    if ((VA.getLocVT() == MVT::i64 && VA.getValVT() == MVT::f32) ||
+        (VA.getLocVT().isInteger() && VA.getValVT() == MVT::f16)) {
```

Comment:
maybe consider bfloat but I guess that's totally busted in globalisel anyway

Reply:
I'll do it as a separate patch. I don't think it belongs in this one.